### ISSUE TITLE
bcc_paracrystal and fcc_paracrystal scaling, docs for all paracrystal (Trac #975)

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasmodels#155]
Following some discussions with Prof. Gustavo González Gaitano, University of Navarra, Spain and some test calculations. 

Given that the docs are already modified for the new "jitter" integrals (in progress??) which branch(es) should I use to fix the below please?

It now seems clear that:

1) The docs for all three paracrystal models talk about "nearest neighbour distances", but these should be lattice spacings, as previously noted in  #284 (used to be ticket 904 in SasView/sasview)

2) the Vlattice corrections mentioned in the docs seem wrong for BCC and FCC, and need to be changed there and in the code in order for fits to have the correct scale parameters.

I think that latticescale should be as for SC, but with the multiplicities in for FCC and BCC thus -

Instead of:
FCC 	const double s1 = dnn*sqrt(2.0);
	const double latticescale = 4.0*sphere_volume(radius/s1);

BCC	const double s1 = dnn/sqrt(0.75);
	const double latticescale = 2.0*sphere_volume(radius/s1);

SC	// NB: 4/3 pi r^3 / dnn^3 = 4/3 pi(r/dnn)^3
	const double latticeScale = sphere_volume(radius/dnn);
	answer *= sphere_form(q, radius, sphere_sld, solvent_sld)*latticeScale;

we need for FCC and BCC:

FCC 	
	const double latticescale = 4.0*sphere_volume(radius/dnn);

BCC	
	const double latticescale = 2.0*sphere_volume(radius/dnn);

and in some other places for IQxy

Migrated from http://trac.sasview.org/ticket/975
```json
{
    "status": "new",
    "changetime": "2017-10-27T17:08:17",
    "_ts": "2017-10-27 17:08:17.935251+00:00",
    "description": "Following some discussions with Prof. Gustavo Gonz\u00e1lez Gaitano, University of Navarra, Spain and some test calculations. \n\nGiven that the docs are already modified for the new \"jitter\" integrals (in progress??) which branch(es) should I use to fix the below please?\n\nIt now seems clear that:\n\n1) The docs for all three paracrystal models talk about \"nearest neighbour distances\", but these should be lattice spacings, as previously noted in #805\n\n2) the Vlattice corrections mentioned in the docs seem wrong for BCC and FCC, and need to be changed there and in the code in order for fits to have the correct scale parameters.\n\nI think that latticescale should be as for SC, but with the multiplicities in for FCC and BCC thus -\n\nInstead of:\nFCC \tconst double s1 = dnn*sqrt(2.0);\n\tconst double latticescale = 4.0*sphere_volume(radius/s1);\n\nBCC\tconst double s1 = dnn/sqrt(0.75);\n\tconst double latticescale = 2.0*sphere_volume(radius/s1);\n\nSC\t// NB: 4/3 pi r^3 / dnn^3 = 4/3 pi(r/dnn)^3\n\tconst double latticeScale = sphere_volume(radius/dnn);\n\tanswer *= sphere_form(q, radius, sphere_sld, solvent_sld)*latticeScale;\n\nwe need for FCC and BCC:\n\nFCC \t\n\tconst double latticescale = 4.0*sphere_volume(radius/dnn);\n\nBCC\t\n\tconst double latticescale = 2.0*sphere_volume(radius/dnn);\n\nand in some other places for IQxy\n",
    "reporter": "richardh",
    "cc": "",
    "resolution": "",
    "workpackage": "SasView Bug Fixing",
    "time": "2017-06-07T16:17:20",
    "component": "sasmodels",
    "summary": "bcc_paracrystal and fcc_paracrystal scaling, docs for all paracrystal",
    "priority": "minor",
    "keywords": "",
    "milestone": "SasView 4.3.0",
    "owner": "richardh",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2017/10/27 17:08:06`: **butler** commented:

These are model docs right?  In that case you can do it directly in master for now (those are in sasmodels not !SasView)

**Comment by butlerpd** [comment]
Trac update at `2017/10/27 17:08:17`: **butler** changed milestone from "SasView 4.2.0" to "SasView 4.3.0"

**Comment by butlerpd** [comment]
I believe all of this has now been resolved, with a roadmap for implementation, in #284.  This ticket is referenced in that ticket already so I think we can now close this ticket. However it might be good if @RichardHeenan took a look at the writeup in #284 to make sure he concurs?

**Comment by RichardHeenan** [comment]
Though I did some of the earlier digging here, it would now take me a considerable time (that I don't really have)  to go back to investigate again, which in any case the other contributors seem to have done very carefully. So I will close this ticket. Would strongly suggest that the discussion in #284 is included as a footnote to the model documentation or in the code itself, to help anyone else going along the same route in the future!  Also strongly agree that sasview would benefit from a route to return derived quantities & intermediates to the user.
